specification/ai/OpenAI.Assistants/messages/models.tsp

```diff
@@ -207,3 +231,182 @@ model MessageFile {
+  delta: MessageDelta;
+}
+
+/** Represents the typed 'delta' payload within a streaming message delta chunk. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDelta {
+  /** The entity that produced the message. */
+  role: MessageRole;
+
+  /** The content of the message as an array of text and/or images. */
+  content: MessageDeltaContent[];
+
+  /** A list of file IDs that the assistant can use. */
+  @encodedName("application/json", "file_ids")
+  fileIds?: string[] = [];
+}
+
+/** The abstract base representation of a partial streamed message content payload. */
+@discriminator("type")
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDeltaContent {
+  /** The index of the content part of the message. */
+  index: int32;
+
+  /** The type of content for this content part. */
+  type: string;
+}
+
+/** Represents a streamed image file content part within a streaming message delta chunk. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDeltaImageFileContent extends MessageDeltaContent {
+  /** The type of content for this content part, which is always "image_file." */
+  type: "image_file";
+
+  /** The image_file data. */
+  @encodedName("application/json", "image_file")
+  imageFile?: MessageDeltaImageFileContentObject;
+}
+
+/** Represents the 'image_file' payload within streaming image file content. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDeltaImageFileContentObject {
+  /** The file ID of the image in the message content. */
+  @encodedName("application/json", "file_id")
+  fileId?: string;
+}
+
+/** Represents a streamed text content part within a streaming message delta chunk. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDeltaTextContentObject extends MessageDeltaContent {
+  /** The type of content for this content part, which is always "text." */
+  type: "text";
+
+  /** The text content details. */
+  text: MessageDeltaTextContent;
+}
+
+/** Represents the data of a streamed text content part within a streaming message delta chunk. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDeltaTextContent {
+  /** The data that makes up the text. */
+  value?: string;
+
+  /** Annotations for the text. */
+  annotations?: MessageDeltaTextAnnotation[];
+}
+
+/** The abstract base representation of a streamed text content part's text annotation. */
+@discriminator("type")
+@added(ServiceApiVersions.vFuturePlaceholder)
+model MessageDeltaTextAnnotation {
+  /** The index of the annotation within a text content part. */
+  index: int32;
+
+  /** The type of the text content annotation. */
+  type: string;
+
+  /** The start index of this annotation in the content text. */
+  @encodedName("application/json", "start_index")
+  startIndex: int32;
+
+  /** The end index of this annotation in the content text. */
+  @encodedName("application/json", "end_index")
+  endIndex: int32;
```

Comment:
Good catch on both counts! I'll fix the optionality on these -- minor backwards breaking change, but it's a descriptive alignment bugfix rather than an actual API modification.

I'll also move/duplicate the properties into the child types. It "feels" relatively safe and the introduction of optionality would also protect us from the worst-case break, but there's really not enough there to justify consolidating relative to the source spec in any case.